src/snowflake/connector/compute_chunk_size.py

```diff
@@ -0,0 +1,69 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+from __future__ import annotations
+
+import logging
+import math
+
+logger = logging.getLogger(__name__)
+
+# Amazon S3 multipart upload limits
+# https://docs.aws.amazon.com/AmazonS3/latest/userguide/qfacts.html
+CURRENT_CHUNK_SIZE = 8 * 1024 ** 2
+MAX_OBJECT_SIZE = 5 * 1024 ** 4
+MAX_PART_SIZE = 5 * 1024 ** 3
+MIN_PART_SIZE = 5 * 1024 ** 2
+MAX_PARTS = 10000
+
+
+class ChunkSizeCalculator:
+    def __init__(
+        self,
+        current_chunk_size: int = CURRENT_CHUNK_SIZE,
+        max_object_size: int = MAX_OBJECT_SIZE,
+        max_part_size: int = MAX_PART_SIZE,
+        min_part_size: int = MIN_PART_SIZE,
+        max_parts: int = MAX_PARTS,
+    ) -> None:
+        self.current_chunk_size = current_chunk_size
+        self.max_object_size = max_object_size
+        self.max_part_size = max_part_size
+        self.min_part_size = min_part_size
+        self.max_parts = max_parts
+
+    def compute_chunk_size(self, file_size: int) -> int:
+        chunk_size = self.current_chunk_size
+        # check if we don"t exceed the allowed max file size 5 TiB
+        if file_size is not None and file_size < self.max_object_size:
+            chunk_size = self._check_max_parts(chunk_size, file_size)
+        else:
+            error_message = f"File size {file_size} exceeds the maximum file size {self.max_object_size}."
+            logger.error(error_message)
+            raise Exception(error_message)
+        return self._check_min_chunk_size(chunk_size)
+
+    # check lower chunk size limit 5 MiB
+    def _check_min_chunk_size(self, current_chunk_size: int) -> int:
+        chunk_size = current_chunk_size
+        if chunk_size < self.min_part_size:
+            logger.debug(
+                f"Setting chunksize to {self.current_chunk_size} instead of the default {current_chunk_size}."
+            )
+            return self.min_part_size
+        else:
+            return current_chunk_size
+
+    # check max chunks number 10k
+    def _check_max_parts(self, current_chunk_size: int, file_size: int) -> int:
+        chunk_size = current_chunk_size
+        num_parts = int(math.ceil(file_size / float(current_chunk_size)))
```

Comment:
we shouldn't need `int` after `ceil`

Reply:
Got rid of the `int` and as agreed the `float` since `current_chunk_size` is an `int`.